backend/analytics_server/mhq/api/deployment_analytics.py

```diff
@@ -62,12 +62,12 @@ def get_team_deployment_analytics(
     pr_filter: PRFilter = apply_pr_filter(
         pr_filter, EntityType.TEAM, team_id, [SettingType.EXCLUDED_PRS_SETTING]
     )
-    code_repo_service = CodeRepoService()
+    repository_service = get_repository_service()
 
-    team_repos: List[TeamRepos] = code_repo_service.get_active_team_repos_by_team_id(
+    team_repos: List[TeamRepos] = repository_service.get_active_team_repos_by_team_id(
         team_id
     )
-    org_repos: List[OrgRepo] = code_repo_service.get_active_org_repos_by_ids(
+    org_repos: List[OrgRepo] = repository_service.get_active_org_repos_by_ids(
```

Comment:
since we are not using team_repos apart from fetching the repos later by ids, we can use PullRequestAnalyticsService.get_team_repos(team_id). This internally calls repo fucns to join the tables and get the same result. Also avoid the hassle

Reply:
RepositoryService does have a similar method but that is only being used in team repo CRUD api, we can keep it that way.